bin/pyenv-installer

```diff
@@ -7,6 +7,17 @@ if [ -z "$PYENV_ROOT" ]; then
   export PYENV_ROOT="${HOME}/.pyenv"
 fi
 
+if [ -z "$shell" ]; then
+  shell="$(ps -p "$PPID" -o 'args=' 2>/dev/null || true)"
+  shell="${shell%% *}"
+  shell="${shell##-}"
+  shell="${shell:-$SHELL}"
+  shell="${shell##*/}"
+  shell="${shell%%-*}"
+fi
```

Comment:
Why the heck did we export `--detect-shell`, to be used with `eval` to boot, if we STILL need this?

Reply:
https://github.com/pyenv/pyenv-installer/pull/137#discussion_r1056935124

Reply:
Considering this example:
Run `curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash` in zsh.
And still make it works

Reply:
That just means that `--detect-shell` needs to do better.

Thanks to `eval`, it cam run code in the same process, evading the "shell of parent of parent" problem.
Just make it output the shell detection code that you've copypasted.